plugins/katello/developers.md

```diff
@@ -229,37 +229,99 @@ Make sure to follow the instructions and update the docs to use the current rele
 
-### Step 6. Pin katello-installer puppet modules for release
+### Step 6. Release puppet modules
 
-On the katello-installer's KATELLO-X.Y branch, we need to pin the modules to a minor release.
+Review the modules for releasing - we should be keeping them up to date when making incompatible changes, but do any need bumps of the major or minor version? If so, do so now.
 
-First, we take the pinned Foreman modules in the Puppetfile from the most current release, e.g. https://github.com/theforeman/foreman-installer/blob/1.9-stable/Puppetfile.  Overwrite the Foreman section in katello-installer's Puppetfile with these locked down versions.
+Go through the [list of managed modules](https://github.com/theforeman/foreman-installer-modulesync/blob/master/managed_modules.yml).
 
-Next, we'll lock down the Katello modules. First, Check out all the puppet modules:
+```sh
+# Review the the changes since the last tag
+tig $(git describe --abbrev=0 --tags)..HEAD
 
-```bash
-for module in candlepin certs capsule common elasticsearch gutterball katello crane service_wait pulp
-do
-  git clone git@github.com:Katello/puppet-${module}.git
-done
-```
+# If it needs a major or minor version bump
+# Note: usually we already bumped it for patch releases but do check!
+bundle exec rake module:bump:major
+bundle exec rake module:bump:minor
 
-Review the modules for releasing - we should be keeping them up to date when making incompatible changes, but do any need bumps of the major or minor version? If so, do so now.
+# Update CHANGELOG.md
+# TODO: modulesync to include github_changelog_generator in Gemfile
+# See https://github.com/skywinder/Github-Changelog-Generator#github-token on using a token to avoid ratelimits
+github_changelog_generator -u theforeman -p $(basename $PWD) --unreleased-only -o new-changelog.md --future-release $(bundle exec rake module:version)
+
+# Now merge new-changelog.md into CHANGELOG.md - do note that it doesn't handle
+# branches so if anything was merged to a stable branch you should manually put
+# it in the correct release. This may also break badly on stable branches.
+
+# Create the release commit
+git add CHANGELOG.md
+VERSION=$(bundle exec rake module:version) && git commit -m "Release $VERSION"
 
-Consider the changes that have been made to the module since the last release, do we need a new release?
-Use [blacksmith](https://github.com/maestrodev/puppet-blacksmith) to push to forge.
+# Create a new tag
+# TODO: bundle exec rake module:tag doesn't create a GPG signed tag
+VERSION=$(bundle exec rake module:version) && git tag -s $VERSION -m "Version $VERSION"
 
-```bash
-rake module:release
+# Push the changes to github
+git push --follow-tags
+
+# Push the release to the forge
+# TODO: Automate this through Travis like voxpupuli?
```

Comment:
https://github.com/voxpupuli/puppet-dhcp/blob/2aaf95a6091f72cf91b8b33883b629714859a7ac/.travis.yml#L41-L51

Reply:
I'd be fine with automating to release easily and often. @stbenjam may have some thoughts given he handled a lot of Katello module releases

Reply:
Auto-releasing on tags sounds great to me!

Reply:
@ekohl In that case, do you want to leave it as a TODO here to tackle later or before this gets merged?